cpu/nrf5x_common/periph/hwrng.c

```diff
@@ -62,3 +70,23 @@ void hwrng_read(void *buf, unsigned int num)
     NRF_RNG->POWER = 0;
 #endif
 }
+
+#else
+
+void hwrng_read(void *buf, unsigned int num)
+{
+    uint32_t ret;
+    uint8_t avail;
+
+    assert(num <= 0xff);
+
+    /* this is not the most efficient, but this way we can assure that there are
+     * enough bytes of random data available */
+    do {
+        sd_rand_application_bytes_available_get(&avail);
+    } while (avail < (uint8_t)num);
+
+    ret = sd_rand_application_vector_get((uint8_t *)buf, (uint8_t)num);
+    assert(ret == NRF_SUCCESS);
```

Comment:
Should also be `(void)ret;`'ed, in case assert resolves to NOP.

Reply:
yes, I have missed that.

@PeterKietzmann I think its the easiest if you fix that in your branch directly?!